cuqi/experimental/mcmc/_sampler.py

```diff
@@ -72,6 +72,14 @@ def validate_target(self):
         """ Validate the target is compatible with the sampler. Called when the target is set. Should raise an error if the target is not compatible. """
         pass
 
+    def _pre_sample(self):
```

Comment:
On whether to make these abstract, what would be the pros and cons? If non-abstract, as now, it seems ok to me that only if the user (coder of a new sampler) specifically wants something to happen there, they will overwrite these methods. Otherwise, the current "pass" methods will simply be called.

Reply:
I am fine with the _pre_sample and _pre_warmup being was they are here (that is not abstract). I can already tell we will use some general implementation, e.g. for "initial_point" in the base class very soon. Then I would suggest calling `super()._pre_warmup()` and `super()._pre_sample()` in the subclass though.

Reply:
Probably it is fine just to make it not abstract. The only advantage of making it abstract in my opinion is to alert the developer to take care of these two methods. Thanks for the input here.

Reply:
LGTM. Jakob  can resolve if he agrees!